src/machine.rs

```diff
@@ -365,31 +387,75 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
         Ok(())
     }
 
+    fn access_local(
+        ecx: &InterpCx<'mir, 'tcx, Self>,
+        frame: &Frame<'mir, 'tcx, Self::PointerTag, Self::FrameExtra>,
+        local: mir::Local,
+    ) -> InterpResult<'tcx, Operand<Self::PointerTag>> {
+        match frame.body.local_decls[local].local_info {
+            mir::LocalInfo::StaticRef { def_id, is_thread_local: true } => {
```

Comment:
Wait is this using *debug info* to affect semantics? I'm afraid I am rather strongly opposed to this approach.

I gather this is related to https://github.com/rust-lang/rust/pull/70598/files#r401254938. I think we can and should solve that issue differently, in a way that does not rely on debug info.

Reply:
> I gather this is related to https://github.com/rust-lang/rust/pull/70598/files#r401254938.

Yes, it is related to that discussion.

> I think we can and should solve that issue differently, in a way that does not rely on debug info.

Do you have a concrete idea that does not require fixing #70685 first? (I am afraid that fixing #70685 will take a very long time.)